cmd/opampsupervisor/e2e_test.go

```diff
@@ -617,77 +617,109 @@ func TestSupervisorConfiguresCapabilities(t *testing.T) {
 }
 
 func TestSupervisorBootstrapsCollector(t *testing.T) {
-	agentDescription := atomic.Value{}
+	tests := []struct {
+		name     string
+		cfg      string
+		env      []string
+		precheck func(t *testing.T)
+	}{
+		{
+			name: "With service.AllowNoPipelines",
+			cfg:  "nocap",
+		},
+		{
+			name: "Without service.AllowNoPipelines",
```

Comment:
This test runs a shell script that masks the Collector's support for the `service.AllowNoPipelines` feature gate and requires it to run with the nop components. I think we can rely only on the feature gate and remove this test at some point.